recipes/libpqxx/all/conanfile.py

```diff
@@ -35,24 +39,35 @@ def configure(self):
         if self.options.shared:
             del self.options.fPIC
 
+    def validate(self):
         compiler = str(self.settings.compiler)
         compiler_version = Version(self.settings.compiler.version.value)
 
-        minimal_version = {
-            "Visual Studio": "15",
-            "gcc": "7",
-            "clang": "6",
+        lib_version_7_5_0_or_above = Version(self.version) >= "7.5.0"
+        minimum_compiler_version = {
+            "Visual Studio": "16" if lib_version_7_5_0_or_above else "15",
```

Comment:
Previously, versions `7.5.x` were being built with Visual Studio 15 (and GCC 7 and Clang 6), right? With this change we are discarding those versions...

Reply:
Yes. libpqxx 7.6.0 no longer supports these versions.

Visual Studio 15 (2017) cannot build because libpqxx 7.6.0 require `[[deprecated]]` attribute support.
However, Visual Studio 15 support is not complete and the build will fail.
https://github.com/jtv/libpqxx/blob/7.6.0/NEWS#L17

MSVC issues:
  * https://developercommunity.visualstudio.com/t/compilation-error-on-deprecated-template-construct/139521
  * https://developercommunity.visualstudio.com/t/deprecated-attribute-with-template-member-function/421223

gcc 6 was already unsupported in libpqxx version 7.5.0 (it could be built, but there was no guarantee that it would work).
https://github.com/jtv/libpqxx/blob/7.6.0/NEWS#L34

I couldn't get clang 6 to work, but it may be possible to build it with the PR (jtv/libpqxx#468) patch.
I'll check it once and see if it works.

Reply:
@jgsogo clang 6 was able to build libpqxx 7.6.0 with patch `fix-inline-constexpr-local-variable-problem-7.6.0.patch`.
Resumed support for clang 6 in the last commit (e695f6181e467fce88ba8d2b442154872f317d0b).

Reply:
I think I didn't manage to explain myself. I meant it should be:

```
    lib_version_7_6_0_or_above = Version(self.version) >= "7.6.0"
    minimum_compiler_version = {
            "Visual Studio": "16" if lib_version_7_6_0_or_above else "15",
```

It is the new version `7.6.0` the one that requires new compiler versions. Old versions (7.5.x series) were _working_ with those compiler versions.

Reply:
> Old versions (7.5.x series) were working with those compiler versions.

The previous version used macros to work around the MSVC 15 bug (jtv/libpqxx#406).
However, in order to continue to provide this workaround using `__declspec(deprecated)`, there was the problem of having to define multiple macros when other C++ standard attributes and `__declspec` wanted to be used for functions or classes.

In this version, the maintainer made the decision to remove this workaround.

Reply:
Yes, but the changes in this PR are dropping support for some compiler versions in `libpqxx/7.5.x`, see the output (https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/6566/16-production/libpqxx/7.5.2//summary.json):

![image](https://user-images.githubusercontent.com/1406456/128683309-96ac19af-a125-4e31-a752-b7045fe165fa.png)

and they were available before:

`conan search libpqxx/7.5.2@ -r conancenter --table=output.html`
![image](https://user-images.githubusercontent.com/1406456/128683469-acc69eae-1f4f-4249-b49b-d1118ab9849d.png)


I'm just saying that **the condition should be _version 7.6.0 or above_** (not 7.5.0 like it is now in this PR).

Reply:
@jgsogo Oops, that was an oversight on my part. Thanks for bringing it to my attention!

Fixed the conditional expression in the last commit (09d2f88f12527be2d21474be2a20cee0726819c8).